src/components/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-console */
+import { Component } from 'react';
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  time: Date;
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    time: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ time: new Date() });
+      console.info(this.state.time.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    if (prevState.time.getSeconds() === this.state.time.getSeconds()) {
```

Comment:
If you press the left mouse button several times, then printing in the debugging of changing the name of the clock does not work correctly.
![image](https://user-images.githubusercontent.com/105992827/203105780-c8211003-fbd1-44e0-bd87-a4c967008a7c.png)

```suggestion
    if (prevProps.name !== this.props.name) {
```

Reply:
thank you for valid comments! :)